satpy/readers/generic_image.py

```diff
@@ -102,8 +102,8 @@ def end_time(self):
 
     def get_dataset(self, key, info):
         """Get a dataset from the file."""
-        logger.debug("Reading %s.", key)
-        return self.file_content[key['name']]
+        logger.debug("Reading 'image' (ignoring provided key %s).", key)
+        return self.file_content['image']
```

Comment:
How about you either:

1. Copy `self.file_content['image']` (with `.copy()` method) and update the attributes with the `info` dictionary. Like `img_copy.attrs.update(info)`. I think this should update the `name` and any other `DataID` information that someone might include in the YAML.
2. Keep the old way of doing `key['name']` but update the `read` method to write the file content to the right `file_content` key.

Option 1 is easiest. Option 2 provides the most flexibility in the future. @pnuu let us know if there is a reason not to move `.read` to the `get_dataset` call as that also seems like it could help here.

Reply:
Let's wait for @pnuu 's input and I'll change it accordingly. Option 2 seems nicer to me.

Reply:
The `'image'` is there just to make things work with the existing way Satpy is designed to load data with a named dataset. In an image, there's only one dataset and `scn.load()` should probably just load the image. But yeah, when loading data with multiple readers that'd not work.

I think option 2 is the better way.

Reply:
I'm going to remove the `read()` completely as it is only used in the ctor and move the reading stuff to `get_dataset()`. Then there's also no need to keep an attribute `self.file_content` any longer. The "How-To for custom readers" (https://satpy.readthedocs.io/en/stable/dev_guide/custom_reader.html) only demands implementing `get_dataset()`. 
And I couldn't find any other readers that use `GenericImageFileHandler` as base class.
Keeping `self.file_content` seems only necessary for caching reasons. But I think there are other mechanisms in satpy that care about this, right?
Okay?

Reply:
Regarding moving the functionality: sounds good as long as the code is clean.

Regarding file_content, that is likely just a common practice used from other readers. It also makes writing tests easier, at least in other readers, because complex NetCDF/HDF files can be described by a dictionary and put in `file_content`. For this reader, maybe that isn't necessary. As long as the tests can be made to work with whatever you do then :+1:

Reply:
I've decided to lease reading in ctor as it is because otherwise `get_area_def()` needs special treatment. The area is set during reading. Although it seems that  `get_area_def()`  is called after `get_dataset()` and therefore area should be already available in my tests - it don't know if it's the case in all situations..
The latest commit introduces a class attribute `dataset_name` that is set to default `image` in the read function. In derived fake readers you can provide own file_content and leave `ds_name` empty to work with multiple datasets - if somebody really wants to do this..

Reply:
FYI there was a semi-recent change in the base reader that changed the order of when the dataset was loaded and when the area was loaded. It used to be that the area was always loaded first (if I remember correctly), but now it is loaded after.